src/covjson_pydantic/domain.py

```diff
@@ -23,34 +22,30 @@ class CompactAxis(BaseModel):
     stop: float
     num: PositiveInt
 
-    @root_validator(skip_on_failure=True)
-    def single_value_case(cls, values):
-        if values["num"] == 1 and values["start"] != values["stop"]:
+    @model_validator(mode="after")
+    def single_value_case(self):
+        if self.num == 1 and self.start != self.stop:
             raise ValueError("If the value of 'num' is 1, then 'start' and 'stop' MUST have identical values.")
-        return values
+        return self
 
 
 ValuesT = TypeVar("ValuesT")
 
 
-class ValuesAxis(GenericModel, Generic[ValuesT]):
-    dataType: Optional[str]  # noqa: N815
-    coordinates: Optional[List[str]]
+class ValuesAxis(BaseModel, Generic[ValuesT]):
```

Comment:
Looking at this again, pydantic `GenericModel` was really intended here, as it this object a local config.

But I think we get the same config if we use based model `CovJsonBaseModel` here, and drop the local config, agreed?